### PR TITLE
Use triton wheel no fork

### PR DESCRIPTION
**Pull request by mengfei-jiang**
## Motivation
  Currently triton is either built from source in CI (`build-triton` job) or relies on whatever version is pre-installed in the Docker base image. This is slow, fragile, and inconsistent across workflows. AMD now publishes pre-built `amd-triton` wheels on [ROCm 7.0](https://pypi.amd.com/triton/rocm-7.0.0/simple/), [ROCm 7.1](https://pypi.amd.com/triton/rocm-7.1.0/simple/), and [ROCm 7.2](https://pypi.amd.com/triton/rocm-7.2.0/simple/), making source builds unnecessary. This PR centralizes triton installation into a single shared script that auto-detects the ROCm version and installs the matching `amd-triton` wheel, ensuring all CI workflows and local development use the same triton distribution.                                                                                                                                                                
                                          
  Additionally, there were no tests verifying that triton operators work correctly with `torch.compile`. This PR adds 10 torch.compile compatibility tests to catch regressions early.                                                                                                                                                              
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    
## Technical Details

  - Replace triton source builds with amd-triton from AMD PyPI: Introduce a shared install_triton.sh script that auto-detects the ROCm version via rocm-core and installs the matching amd-triton wheel from https://pypi.amd.com/triton/rocm-{major}.{minor}.0/simple/. This eliminates the need to build triton from source in CI, removing the build-triton job from triton-test.yaml (~80 lines).                                                                                                                                                                                                                   
  - Unify triton installation across all CI workflows: Add install_triton.sh to aiter-test.yaml, atom-test.yaml, sglang_downstream.yaml, and vllm_benchmark.yaml, ensuring all workflows use the same amd-triton version. The script uninstalls all conflicting triton variants (triton, pytorch-triton, pytorch-triton-rocm, triton-rocm, amd-triton) before installing.                                                                                                                                                                                                                                             
  - Auto-install in develop mode: setup.py now calls install_triton.sh during python setup.py develop, so developers get amd-triton installed automatically.
  - Add torch.compile compatibility tests: 10 new test files under op_tests/triton_tests/torch_compile/ verifying that triton operators work correctly with torch.compile(backend="inductor", fullgraph=True). Covers activation, fused_mul_add, gemm, moe_routing, quantization (per-tensor/per-token), rmsnorm, rope, softmax, and topk.                                                                                                                                                                                                                                                                            
  - Deduplicate _get_compiled helper: Extract the shared _get_compiled function into torch_compile/__init__.py so all test files import from a single location.                                                                                                                                                
  - Add torch_compile test times to split_tests.sh: Include FILE_TIMES for the 10 new tests (~90s total) to enable proper shard balancing in CI.                                                                                                                                                               
  - Update README: Add Triton section documenting amd-triton installation with AMD PyPI index URLs and the install_triton.sh script.        

## Test Plan

  - triton-test.yaml passes without the build-triton job (amd-triton installed via install_triton.sh)                                                                                                                                                                                                          
  - aiter-test.yaml standard and multi-gpu tests pass with Install amd-triton step
  - atom-test.yaml, sglang_downstream.yaml, vllm_benchmark.yaml CI workflows pass                                                                                                                                                                                                                              
  - All 10 torch_compile tests  on MI300X and MI35X runners                                                                                                                                                                                                                                                

## Test Result

All can pass

## Submission Checklist

- [x] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

### CODE REVIEW

**Comment by brunomazzottiamd** [comment]
**Question:** How much time the new Triton tests in `op_tests/triton_tests/torch_compile` take to execute (in MI300X and MI35XX runners)?

Please be mindful about the CI time. We have recently made a great effort to reduce the time taken by Triton Test jobs.

**Comment by brunomazzottiamd** [comment]
I can see that

```python
def _get_compiled(fn):
    return torch.compile(
        fn, backend="inductor", fullgraph=True, options={"max_autotune": True}
    )
```

function has duplicates on every test file.

Is it possible to have a single implementation imported by the test files?

**Comment by mengfei-jiang** [comment]
> **Question:** How much time the new Triton tests in `op_tests/triton_tests/torch_compile` take to execute (in MI300X and MI35XX runners)?
> 
> Please be mindful about the CI time. We have recently made a great effort to reduce the time taken by Triton Test jobs.

Done

**Comment by mengfei-jiang** [comment]
> I can see that
> 
> ```python
> def _get_compiled(fn):
>     return torch.compile(
>         fn, backend="inductor", fullgraph=True, options={"max_autotune": True}
>     )
> ```
> 
> function has duplicates on every test file.
> 
> Is it possible to have a single implementation imported by the test files?

Done, now it is a single implementation.

**Comment by brunomazzottiamd** [comment]
I've checked the new test files, LGTM! However, I don't have enough CI knowledge to comment on the scripts and workflows. Let's wait for an approaval from AITER CI team.

**Comment by brunomazzottiamd** [comment]
Failures in **Flash Attention Integration** jobs are being addressed in https://github.com/ROCm/aiter/pull/2695.

**Comment by brunomazzottiamd** [comment]
Added `ci:triton-300x` label to trigger execution of Triton unit tests in `gfx942` nodes.

**Review by Dewei-Wang-sh — APPROVED** [review]
overall, lgtm

**Comment by Dewei-Wang-sh** [comment]
> ## Motivation
> Replace the triton dependency management with auto-detection of the ROCm version to install the correct amd-triton 

however, this description is outdated and only show one point, please change accordingly.
you need to clarify what this pr does and what this pr for..

**Review by valarLip — APPROVED** [review]
